Make (Section/Flat)Grids allow delegated props on TS

Grohden opened the pull request:
Extended the SectionGrid and FlatGrid props declarations from the original declarations,
I've also fixed a type issue with SectionGrid prop `sections`, that was declared as a single item instead of list

This should close #95

Grohden commented:
I've made the changes based on the readme, I pasted them on my project and they seem fine. But still I would like a confirmation from the author that I omited all the overriden/changed props from the original components

Grohden commented:
Looking at the readme again, shouldn't the `sections` prop be declared as:
```typescript

export type SectionItem<ItemType> = {
   title: string;
   data: ItemType[];
}

// ...declarations
sections:  SectionItem<ItemType>[];
//...declarations
```
?
Also, not sure, but sections may contain different items for their lists.. but probably TS will infer a union for the data prop of the section item

saleel commented on index.d.ts:
```diff
@@ -6,12 +6,23 @@ import {
  * React Native Super Grid Properties
  */
-export interface FlatGridProps<ItemType = any> {
+export interface FlatGridProps<ItemType = any> extends FlatListAllowedProps<ItemType> {
```
If we can extend from FlatListProps then there is no need to explicitly mention props like `ListEmptyComponent`, `onEndReached`...etc here

Grohden replied:
I've been reviewing the types now, and there are some diffs that seems wrong to me like:
`keyExtractor?: (item: ItemType[], index: number) => string;` - declared on the super grid
`keyExtractor?: (item: ItemT, index: number) => string;` - declared on the rn grid
I'll keep the RN as a source of truth and remove the types that have the same signature or seems to have a wrong one from the lib

saleel commented:
> Looking at the readme again, shouldn't the `sections` prop be declared as:
> 
> ```ts
> export type SectionItem<ItemType> = {
>    title: string;
>    data: ItemType[];
> }
> 
> // ...declarations
> sections:  SectionItem<ItemType>[];
> //...declarations
> ```
> 
> ?
> Also, not sure, but sections may contain different items for their lists.. but probably TS will infer a union for the data prop of the section item

This looks correct. Can you also check how they are doing it in the official definition (of SectionList) and update here accordingly. Please test it locally to ensure both SectionGrid and FlatGrid works with the props mentioned in the Readme and the ones provided by FlatList/SectionList

Grohden commented:
@saleel I've made *some* changes on the types:
* moved common props to a single type
* removed the redeclared props (the ones you mentioned)
* I added the `rowIndex` as a exposed prop on the `renderItem`

I've tested this locally and everything seems fine

saleel commented:
Merged. Thanks
Will release new version shortly
